### PR TITLE
Preparations for PR #5401

### PR DESCRIPTION
**Pull request by wm4**
These commits are part of PR #5401, but contain only minor/harmless things. This is the extent to which I'm fine with splitting this. It's simply an inherently monolithic change.

CC @haasn for the mp_image metadata thing

### CODE REVIEW

**Comment by haasn** [comment]
> CC @haasn for the mp_image metadata thing

w.r.t the `color.light`, it's normally just used to distinguish between HLG (which is scene-referred) and other (display-referred) color spaces; and no file tags exist for it. So if you insert an AVFilter that converts from HLG to PQ or something like that (does such a thing even exist?), then what *should* happen to the `color.light` field would sort of be determined by what said filter did. But if we assume such a filter is doing the right thing, it would apply the HLG OOTF as part of the conversion process - so if we re-infer the `color.light` from the colorspace after every roundtrip through AVFrame, we shouldn't run into any issues in practice.

The one open question would be what happens when the user tries using `--vf=format` to overwrite the metadata. If the user's change gets overwritten after the first roundtrip through swscale or whatever, that would be weird and unexpected. But to be honest, maybe it's time to replace `--vf=format` by dedicated options for overwriting the metadata? Using `vf_format` for that always seemed like a hack to me, because `vf_format` on the one hand is supposed to convert between formats without changing the interpretation of the pixels (e.g. when changing the color representation or color system) - but on the other hand, it changes the interpretation without affecting the pixels (e.g. when changing the primaries or gamma).

**Comment by wm4** [comment]
So why is `light` a mp_image field at all, instead of just being determined in the renderer or so?

File: video/img_format.c [hunk 1]
```diff
@@ -486,6 +486,20 @@ int mp_imgfmt_select_best(int dst1, int dst2, int src)
     return pixfmt2imgfmt(avcodec_find_best_pix_fmt_of_list(dstlist, srcpxf, 1, 0));
 }
 
+// Same as mp_imgfmt_select_best(), but with a list of dst formats.
+int mp_imgfmt_select_best_list(int *dst, int num_dst, int src)
```
[the remark this answers is not in the document]
Comment (wm4) [reply video/img_format.c]: Yes. Also Libav doesn't have it.

File: options/m_config.c [hunk 2]
```diff
@@ -1437,8 +1437,10 @@ bool m_config_is_in_group(struct m_config *config,
 void *mp_get_config_group(void *ta_parent, struct mpv_global *global,
                           const struct m_sub_options *group)
 {
-    assert(ta_parent); // without you'd necessarily leak memory
     struct m_config_cache *cache = m_config_cache_alloc(ta_parent, global, group);
+    // Make talloc_free(cache->opts) free the entire cache.
```
[the remark this answers is not in the document]
Comment (wm4) [reply options/m_config.c]: Yes. changed.

**Comment by wm4** [comment]
Fixed the commented issues, `light` thing unchanged.

**Comment by haasn** [comment]
> So why is light a mp_image field at all, instead of just being determined in the renderer or so?

Because it technically makes sense to have e.g. scene-referred linear light content being stored in a file. It makes sense to be part of the same colorspace struct that defines colorspaces in general, since it's also a property of the colorspace just in the way as the primaries or the transfer function are.

The only way it doesn't make sense is on a technical level because nothing except `-vf=format` allows you to actually change it. In theory we could remove it from mp_image and instead introduce an extra option affecting the VO directly. In theory, we could do that for all of the colorspace overrides too (see: my aforementioned distaste for having `vf_format` handle different tasks). I wouldn't mind much either way - it just sort of makes ‘sense’ where it is currently, but that alone is not a strong reason to keep it there if it causes problems due to the missing AVFilter interface.

**Pull request merged**